src/main/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAligner.scala

```diff
@@ -147,13 +173,40 @@ class NeedlemanWunschAligner(val scoringFunction: (Byte,Byte) => Int,
     * @return an [[Alignment]] object representing the alignment
     */
   protected def generateAlignment(query: Array[Byte], target: Array[Byte], scoring: Matrix[Int], trace: Matrix[Direction]): Alignment = {
-    var i = query.length
-    var j = target.length
     var currOperator: CigarOperator = null
     var currLength: Int = 0
     val elems = new mutable.ArrayBuffer[CigarElem]()
 
-    while (i != 0 || j != 0) {
+    // Always start at the end of the query
+    var i = query.length
+
+    // For the target sequence, start at the end for Global, or the highest scoring cell in the last row for Glocal
+    var j = this.mode match {
+      case Global =>
+        target.length
```

Comment:
My style, which is awesome, is to put this on the previous line, so it looks like:
```
case Global => target.length
case Glocal =>
  var ...
```
It's like having a `return` statement at the top of your function, it looks clearer to me.

Reply:
Ha!  I go back and forth, and generally try to keep all the `case`s the same style, but I'm happy to change it.